pandas/core/internals/managers.py

```diff
@@ -906,14 +911,25 @@ def fast_xs(self, loc):
 
         # unique
         dtype = _interleaved_dtype(self.blocks)
+
         n = len(items)
-        result = np.empty(n, dtype=dtype)
+        if is_extension_array_dtype(dtype):
+            # we'll eventually construct an ExtensionArray.
+            result = np.empty(n, dtype=object)
```

Comment:
Do people find this confusing? I can either

1. duplicate the for loop, using `list.append` for EAs and inserting into `result` for other
2. use lists everywhere
3. use this

I chose this implementation because I assume it's slightly for wide dataframes with a numpy type, compared to building a list an then `np.asarray(result)` at the end.

Reply:
This implementation looks good to me